### PR TITLE
Support SFT, RM and PPO for text-video-image to text models

### PR DESCRIPTION
**Pull request by htlou**
# Description

Describe your changes in detail.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #1314520` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-Alignment/align-anything/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/PKU-Alignment/align-anything/blob/HEAD/.github/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: align_anything/configs/template.py [hunk 1]
```diff
@@ -467,6 +467,99 @@ def format_sample(self, raw_sample: dict[str, Any]) -> dict[str, Any]:
...
+            "video": video_info,
+        }
+        return return_dict  
+    
+@register_template('NExTQA_preference')
+class NExTQA_PREF:
+    system_prompt: str = '<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n'
+    user_prompt: str = '<|im_start|>user\n{input}<|im_end|>\n'
+    assistant_prompt: str = '<|im_start|>assistant\n{output}'
+    split_token: str = '\n'
```
Comment (Gaiejj) [inline align_anything/configs/template.py:512]: Can it correctly seperate sentence in the evaluation of reward modeling?

File: align_anything/trainers/tiv_to_t/sft.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def main():
+    # setup distribution training
+    deepspeed.init_distributed()
+    current_device = get_current_device()
+    torch.cuda.set_device(current_device)
+
+    # read default configs from the yaml file
+    task = os.path.join('tiv_to_t', 'sft')
```
Comment (Gaiejj) [inline align_anything/trainers/tiv_to_t/sft.py:66]: maybe text_video_to_text?

File: README.md [hunk 3]
```diff
@@ -74,6 +74,7 @@ We have a roadmap for future development work `align-anything`:
 
 # News
 
+- 2024-09-12 🔥 We have supported SFT, DPO and PPO for text-video-image to text models!
```
Comment (XuyaoWang) [inline README.md:77]: Add the corresponding section to [README_zh-CN.md](https://github.com/PKU-Alignment/align-anything/blob/main/README_zh-CN.md)

**Comment by htlou** [comment]
Fixed all the comments in the commits above, thus request for another round of review.

File: projects/text_image_video_to_text/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
+# Qwen2-VL Model Doc for Align-Anything
+
+## Environment setup
+
+Currently, the Transformer in pypi is not updated to support the latest Qwen2-VL model, so we need to install the forked stable version of the repo by running:
+pip install git+https://github.com/huggingface/transformers.git
```
Comment (Gaiejj) [inline projects/text_image_video_to_text/README.md:6]:
```bash
pip install git+https://github.com/huggingface/transformers.git
```

File: projects/text_image_video_to_text/README.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+Add a script named `sft_text_image_video_to_text.sh` under the `scripts` file like this:
+
+```bash
+# You can replace it with a local model path
+MODEL_NAME_OR_PATH="Qwen/Qwen2-VL-7B-Instruct"
+# You can replace it with a local dataset path
+TRAIN_DATASETS="../../data"
+TRAIN_DATA_FILE="text_image_video_to_text.json"
+# You can replace it with a new path
+OUTPUT_DIR="../outputs/sft_qwen2_vl"
```
Comment (Gaiejj) [inline projects/text_image_video_to_text/README.md:25]: This is the simplest example we provide. You can refer to their format and replace them with your own dataset.

File: projects/text_image_video_to_text/README.md (hunk 4 above)
Comment (htlou) [reply projects/text_image_video_to_text/README.md:7]: Fixed in https://github.com/PKU-Alignment/align-anything/pull/68/commits/822f056c556cb35520f79ec5742e9b0b7301dd06

File: projects/text_image_video_to_text/README.md (hunk 5 above)
Comment (htlou) [reply projects/text_image_video_to_text/README.md:27]: Added in https://github.com/PKU-Alignment/align-anything/pull/68/commits/04a007fefd4b5bd14cc363083646917736eec893

File: data/mllm_demo/17b7599e-679d-45f9-9166-ad4abbf0d027.mp4
Comment (Gaiejj) [inline data/mllm_demo/17b7599e-679d-45f9-9166-ad4abbf0d027.mp4:1]: Please change the file name

File: data/text_image_video_to_text.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        "video_path": "/aifs4su/yaodong/projects/hantao/remote/align-anything/data/mllm_demo/17b7599e-679d-45f9-9166-ad4abbf0d027.mp4"
```
Comment (Gaiejj) [inline data/text_image_video_to_text.json:15]: absolute path

File: align_anything/configs/template.py [hunk 7]
```diff
@@ -466,6 +466,99 @@ def format_sample(self, raw_sample: dict[str, Any]) -> dict[str, Any]:
         }
         return super().format_example(raw_sample_upd)
 
+@register_template('TIV2T')
+class TIV2T:
+    system_prompt: str = '<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n'
+    user_prompt: str = '<|im_start|>user\n{input}<|im_end|>\n'
+    assistant_prompt: str = '<|im_start|>assistant\n{output}'
+    split_token: str = '\n'
+    separator: str = 'assistant\n'
```
Comment (Gaiejj) [inline align_anything/configs/template.py:475]: the `split_token` should be the same as the `separator`

**Pull request closed without merging**